### PR TITLE
Tests: Level control of account metrics

### PR DESCRIPTION
**Pull request by marki1an**
Tests for https://github.com/prebid/prebid-server-java/pull/1462

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/model/config/AccountMetricsVerbosityLevel.groovy [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.prebid.server.functional.model.config
+
+enum AccountMetricsVerbosityLevel {
+
+    none, basic, detailed
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/model/config/AccountMetricsVerbosityLevel.groovy:5]: Yeah, but we can change this when the core enum will be changed to uppercase.

[the remark this answers is not in the document]
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/model/config/AccountMetricsVerbosityLevel.groovy:5]: We can, but if you look at other enums, you will see that all the test enums are in uppercase. Please be consistent with other code.

File: src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        and: "Account in the DB"
+        def accountId = bidRequest.site.publisher.id
+        def accountMetricsConfig = new AccountConfig(metrics: new AccountMetricsConfig(verbosityLevel: none))
+        def account = new Account(uuid: accountId, config: accountMetricsConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "account.<account-id>.* shouldn't be exists"
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy:28]: "account.<account-id>.* metric shouldn't be populated"

File: src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        def metrics = defaultPbsService.sendCollectedMetricsRequest()
+        assert !metrics.find(it -> { it.key.startsWith("account.${accountId}") })
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy:30]: `metrics.find { it.key.startsWith("account.${accountId}") }`

File: src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        def accountMetricsConfig = new AccountConfig(metrics: new AccountMetricsConfig(verbosityLevel: basic))
+        def account = new Account(uuid: accountId, config: accountMetricsConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "account.<account-id>.requests should be updated"
+        def metrics = defaultPbsService.sendCollectedMetricsRequest()
+        assert metrics["account.${accountId}.requests" as String] == 1
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy:48]: Please make sure that other account metrics are not populated.

**Review by Net-burst — CHANGES_REQUESTED** [review]

File: src/test/groovy/org/prebid/server/functional/model/config/AccountMetricsVerbosityLevel.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/model/config/AccountMetricsVerbosityLevel.groovy:5]: Yeah, you are right, we can do this.

File: src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy [hunk 5]
```diff
@@ -191,4 +196,75 @@ class AuctionSpec extends BaseSpec {
...
+        def accountMetricsConfig = new AccountConfig(metrics: new AccountMetricsConfig(verbosityLevel: none))
+        def account = new Account(uuid: accountId,config: accountMetricsConfig )
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "account.<account-id>.* shouldn't be exists"
+        def metrics = defaultPbsService.sendCollectedMetricsRequest()
+        assert !metrics["account.${accountId}.adapter.generic.bids_received"    as String]
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy]: Done!

File: src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import org.prebid.server.functional.model.config.AccountMetricsConfig
+import org.prebid.server.functional.model.db.Account
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.response.auction.BidResponse
+
+import static org.prebid.server.functional.model.config.AccountMetricsVerbosityLevel.BASIC
+import static org.prebid.server.functional.model.config.AccountMetricsVerbosityLevel.DETAILED
+import static org.prebid.server.functional.model.config.AccountMetricsVerbosityLevel.NONE
+
+class MetricsSpec extends BaseSpec {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/MetricsSpec.groovy:13]: I recommend flushing the metrics before you enter this spec just to be sure that nothing will dirty your tests.

**Review by Net-burst — DISMISSED** [review]
LGTM with minor remark

**Review by Net-burst — APPROVED** [review]
LGTM

**Pull request merged**